libmscore/instrument.cpp

```diff
@@ -443,9 +438,13 @@ Channel::~Channel()
 
 void Channel::firePropertyChanged(Channel::Prop prop)
       {
-      QList<ChannelListener *> list(listeners);
+      QList<QPointer<QObject>> list(listeners);
```

Comment:
If I am correct, you store `ChannelListener` descendants but in order to be stored they have to be inherited from `QObject`? Wouldn't it be better to make `ChannelListener` be derived from `QObject` and have a list of `QPointer`s to `ChannelListener` so that types correctness could be checked at compile-time?
The second question is whether it is actually better to use `QPointer` (as far as I see you had some other solution based on `shared_ptr`?)

Reply:
As far as I can tell, QPointers need to store QObjects.  Qt does have a mechanism for using interfaces, but I am having some trouble getting it to work:
https://forum.qt.io/topic/96845/how-to-cast-interfaces-to-qobjects/5

I had a few attempts to do this with std::shared_ptr but ran into the problem that if you start using shared_ptr, every instance of that object needs to be a shared_ptr which I cannot guarantee.  The existing QPointer system made more sense since all QObjects automatically work with that system.

I would like to use ChannelListener for type correctness, but don't know how to do that with the QPointer objects.  Using QObjects and casting them works to solve the crash.  I'm not sure if I can get this working with the interface object or not.